sandbox/modules/identity/src/internalClusterTest/java/org/opensearch/identity/HttpSmokeTestCaseWithIdentity.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity;
+
+import com.carrotsearch.randomizedtesting.annotations.ThreadLeakScope;
+import org.opensearch.common.network.NetworkModule;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.plugins.Plugin;
+import org.opensearch.test.OpenSearchIntegTestCase;
+import org.opensearch.transport.Netty4ModulePlugin;
+import org.opensearch.transport.nio.MockNioTransportPlugin;
+import org.opensearch.transport.nio.NioTransportPlugin;
+import org.junit.BeforeClass;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+// TODO not sure why ThreadLeakScope.NONE is required
+@ThreadLeakScope(ThreadLeakScope.Scope.NONE)
+public abstract class HttpSmokeTestCaseWithIdentity extends OpenSearchIntegTestCase {
```

Comment:
for my understanding, we should add a description of why we need this class instead of using OpenSearchIntegTestCase directly for BasicAuthenticationIT.

Reply:
I was running into issues where the `SecurityRestFilter` (used in the `getRestHandlerWrapper` extension point) was not getting invoked when running requests with `OpenSearchIntegTestCase`. This class overrides `addMockHttpTransport` to set it to false, adds IdentityPlugin and enables the module.